auth_saml/models/res_users.py

```diff
@@ -55,7 +55,14 @@ def _auth_saml_signin(self, provider: int, validation: dict, saml_response) -> s
             user_saml.with_env(new_env).write({"saml_access_token": saml_response})
 
         if validation.get("mapped_attrs", {}):
-            user.write(validation.get("mapped_attrs", {}))
+            # Only write field that changes to avoid generating Security Update on users
+            # when login/email changes (from mail module)
+            vals = {}
+            for key, value in validation.get("mapped_attrs", {}).items():
+                if not isinstance(value, str) or getattr(user, key) != value:
```

Comment:
```suggestion
                if not isinstance(value, str) or user[key] != value:
```
is it granted that `value` will be compatible w/ the field type? Eg: what if you have a boolean field and you compare `False` with `0`?

Reply:
The mapping as implemented match attributes in the SAML response. I do not believe it can be a type different than string. In keycloak, that does not seem possible.

It’s possible to map one of the attribute to a boolean field. It might be strange to the users that any value is considered as True ("0", "1", "False", etc.) and only "" would put False in the field.
Other type of fields can also be written, but in most cases, that will fail. Writing a many2one with a string fails for example.

To make the code more robust, I’m going to only compare if the value is a string. That will avoid rewriting login/email and any string or selection field which is probably the most often written field when the feature is used.

Reply:
Thanks for the feedback. Can you please report this comment in the code so that we'll have clear where we come from? :)

Reply:
I’ve already adjusted code and added a comment.

https://github.com/OCA/server-auth/pull/726/files#diff-3f2fc558f32a584e17b5ab27d08e9fe0d111070b3230b58d1db2ddc24376309fR62